src/Definition/Oracle.php

```diff
@@ -0,0 +1,231 @@
+        }
+        return $definition;
+    }
+
+    /**
+     * @return array
+     */
+    public function toArray()
+    {
+        return [
+            "type" => $this->getType(),
+            "length" => $this->getLength(),
+            "nullable" => $this->isNullable()
+        ];
+    }
+
+    /**
+     * @param string $type
+     * @throws InvalidTypeException
+     */
+    private function validateType($type)
+    {
+        if (!in_array(strtoupper($type), $this::TYPES)) {
+            throw new InvalidTypeException("'{$type}' is not a valid type");
+        }
+    }
+
+    /**
+     * @param string $type
+     * @param string|null $length
+     * @throws InvalidLengthException
+     */
+    private function validateLength($type, $length = null)
+    {
+        $valid = true;
+        switch (strtoupper($type)) {
+            case "CHAR":
+            case "NCHAR":
+                if (is_null($length) || $length === "") {
+                    break;
+                }
+                if (!is_numeric($length)) {
+                    $valid = false;
+                    break;
+                }
+                if ((int)$length <= 0 || (int)$length > 255) {
+                    $valid = false;
+                    break;
+                }
+                break;
+            case "VARCHAR":
+            case "VARCHAR2":
+            case "NVARCHAR":
+            case "NVARCHAR2":
+                if (is_null($length) || $length === "" || !is_numeric($length)) {
+                    $valid = false;
+                    break;
+                }
+                if ((int) $length <= 0 || (int) $length > 4000) {
+                    $valid = false;
+                    break;
+                }
+                break;
+            case "CLOB":
+            case "NCLOB":
+                if (is_null($length) || $length == "" || !is_numeric($length)) {
+                    $valid = false;
+                    break;
+                }
+                if ((int) $length <= 0 || (int) $length > 4294967295) {
+                    $valid = false;
+                    break;
+                }
+                break;
+            case "LONG":
+                if (is_null($length) || $length === "" || !is_numeric($length)) {
+                    $valid = false;
+                    break;
+                }
+                if ((int) $length <= 0 || (int) $length > 2147483647) {
+                    $valid = false;
+                    break;
+                }
+                break;
+            case "DATE":
+                if (is_null($length) || $length === "") {
+                    break;
+                }
+                if (!is_numeric($length)) {
+                    $valid = false;
+                    break;
+                }
+                if ((int) $length < 0 || (int) $length > 6) {
+                    $valid = false;
+                    break;
+                }
+                break;
```

Comment:
I don't think we should be using DATE lengths as they are byte-lengths and the DATE type always is 6 bytes.   It's not a variable amount.  I suppose we can accept the length parameter, since it is in the metadata, but I don't think that a statement such as `CREATE TABLE "whatever" (date_column DATE(2))` is possible.